### DISCUSSION TITLE
questions on monai labels

### DISCUSSION THREAD

**Discussion by joepareti54** [Project-MONAI/MONAILabel#1137] [Q&A]
is 'monai label' an AI model that needs to be trained to support users' annotations? What kind of model?

is the output of monai labels a dataset with labels that will be used to train ANOTHER ML model ?

**Comment by danieltudosiu** [comment]
Hi Joe,

MONAI Label is a framework for training and using interactive segmentation networks. 

An interactive segmentation network is a network which can accept signals (clicks, strokes or bounding boxes) from the user and predict a segmentation mask. 

At the moment, the network that is trained can be any network that you wish to code or just drop in from MONAI Core.  This is due to the user's signal being passed as additional channels with the imaging input.

As far as I am aware MONAI Label at the moment also supports Active Learning strategies for training said network which prioritises what sample should be picked based on epistemic (dropout uncertainty) and aleatoric (test time augmentation uncertainty) uncertainties.

The output of MONAI Label can be considered a segmentation network that does a good job at segmenting the target of interest, BUT it can ALSO accept the user's signal to IMPROVE said segmentation. Thus, it would speed up the generation of golden standard segmentation (human-based segmentations).

As I am a new developer of MONAI Label so @diazandr3s please let me know if I missed anything.

Cheers,

Dan

**Comment by tangy5** [comment]
Hi @joepareti54 , for simpler answer.

MONAI Label is a platform for deploying AI models, and connect to annotation tool so that users can use AI model using viewer's UI. 

This link is recommended to get started with MONAI Label.

We have some sample models and sample apps, such as spleen segmentation, multi-organ segmentation, BRATS segmentation, etc..
Also pathology and endoscopy applications. 

Thank you for the interests in MONAI Label, which is an open-source project and welcome developers and users!

Cheers

**Reply by joepareti54** [reply]
thank you so much for your answers. To explain the background to my question: my only experiene with image segmentation has been an Nvidia workshop aimed at segmenting the heart ventricle. To do that there was a dataset with labels and a NN that was trained using that dataset. So my understanding of monai labels is something that delivers the labels in a form that can be used to train my model. Is that correct? you said monai labels accpts scribbles, so the mapping between scribbles and labels requires an additional ML model ?

**Reply by diazandr3s** [reply]
Thanks for asking this, @joepareti54.

There have been great answers already :)

To this question:

> So my understanding of monai labels is something that delivers the labels in a form that can be used to train my model. Is that correct?

You could define and then train your own model in MONAI Label. 

There is the concept of apps (https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps) which is a set of deep learning models (https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/radiology#list-of-pretrained-models) that you can use to create an annotated dataset. 

If say, for instance, you have a CT or MR dataset to annotate. You could start using the radiology app with the segmentation model. This means you'll train this UNET https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation.py#L69

However, as @danieltudosiu said, you're not limited to that network. You could use out of the box any of the available networks in MONAI Core: https://github.com/Project-MONAI/MONAI/tree/dev/monai/networks/nets

In the end, the result is an annotated dataset and a trained model. All the apps in MONAI Label are based on MONAI core library: https://github.com/Project-MONAI/MONAI

> you said monai labels accpts scribbles, so the mapping between scribbles and labels requires an additional ML model ?

Scribbles is one of the available functionalities in MONAI Label to create annotations from scratch. More information about the scribbles functionality is here: https://github.com/Project-MONAI/MONAILabel/wiki/Scribbles

BTW, here is a workshop we did recently: https://youtu.be/wtiEe_jiUzg?t=2239
And here is a deep dive series for MONAI Label: https://www.youtube.com/watch?v=8y1OBQs2wis&list=PLtoSVSQ2XzyD4lc-lAacFBzOdv5Ou-9IA

Hope this helps
